classes/settings_provider.php

```diff
@@ -261,12 +274,50 @@ public static function add_seb_config_file(\mod_quiz_mod_form $quizform, \Moodle
     }
 
     /**
-     * Add SEB settings elements.
+     * Add Suppress Safe Exam Browser download button.
+     *
+     * @param \mod_quiz_mod_form $quizform the quiz settings form that is being built.
+     * @param \MoodleQuickForm $mform the wrapped MoodleQuickForm.
+     */
+    protected static function add_seb_suppress_download_link(\mod_quiz_mod_form $quizform, \MoodleQuickForm $mform) {
+        if (self::can_change_seb_suppresssebdownloadlink($quizform->get_context())) {
+            $element = $mform->createElement('selectyesno',
+                'seb_suppresssebdownloadlink',
+                get_string('seb_suppresssebdownloadlink', 'quizaccess_seb')
+            );
+            self::insert_element($quizform, $mform, $element);
+            self::set_type($quizform, $mform, 'seb_suppresssebdownloadlink', PARAM_BOOL);
+            self::set_default($quizform, $mform, 'seb_suppresssebdownloadlink', 0);
+            self::add_help_button($quizform, $mform, 'seb_suppresssebdownloadlink');
+        }
+    }
+
+    /**
+     * Add Allowed Browser Exam Keys setting.
+     *
+     * @param \mod_quiz_mod_form $quizform the quiz settings form that is being built.
+     * @param \MoodleQuickForm $mform the wrapped MoodleQuickForm.
+     */
+    protected static function add_seb_allowedbrowserexamkeys(\mod_quiz_mod_form $quizform, \MoodleQuickForm $mform) {
```

Comment:
As above, should all of these individual elements check for is_conflicting_permissions ?

Reply:
I understand it is now only for the top level USE  case.